go/libraries/doltcore/sqle/commit_diff_table.go

```diff
@@ -0,0 +1,397 @@
+	}
+
+	next := itr.partitions[itr.i]
+	itr.i++
+
+	return next, nil
+}
+
+func (itr *SliceOfPartitionsItr) Close() error {
+	return nil
+}
+
+func (dt *CommitDiffTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error) {
+	if dt.requiredFilterErr != nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), dt.requiredFilterErr)
+	} else if dt.toCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneToCommit)
+	} else if dt.fromCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneFromCommit)
+	}
+
+	toRoot, toName, toDate, err := dt.rootValForFilter(ctx, dt.toCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromRoot, fromName, fromDate, err := dt.rootValForFilter(ctx, dt.fromCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	toTable, _, err := toRoot.GetTable(ctx, dt.name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromTable, _, err := fromRoot.GetTable(ctx, dt.name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSliceOfPartitionsItr([]sql.Partition{diffPartition{
+		to:       toTable,
+		from:     fromTable,
+		toName:   toName,
+		fromName: fromName,
+		toDate:   toDate,
+		fromDate: fromDate,
+	}}), nil
+}
+
+func (dt *CommitDiffTable) rootValForFilter(ctx *sql.Context, eqFilter *expression.Equals) (*doltdb.RootValue, string, *types.Timestamp, error) {
+	var hashStr string
+	var ok bool
+	if _, isGetField := eqFilter.BinaryExpression.Left.(*expression.GetField); !isGetField {
+		val, err := eqFilter.BinaryExpression.Left.Eval(ctx, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		if hashStr, ok = val.(string); !ok {
+			return nil, "", nil, fmt.Errorf("received '%v' when expecting commit hash string", val)
+		}
+	} else if _, isGetField = eqFilter.BinaryExpression.Right.(*expression.GetField); !isGetField {
+		val, err := eqFilter.BinaryExpression.Right.Eval(ctx, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		if hashStr, ok = val.(string); !ok {
+			return nil, "", nil, fmt.Errorf("received '%v' when expecting commit hash string", val)
+		}
+	}
+
+	var root *doltdb.RootValue
+	var commitTime *types.Timestamp
+	if strings.ToLower(hashStr) == "working" {
+		root = dt.workingRoot
+	} else {
+		cs, err := doltdb.NewCommitSpec(hashStr)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		cm, err := dt.ddb.Resolve(ctx, cs, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		root, err = cm.GetRootValue()
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		meta, err := cm.GetCommitMeta()
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		t := meta.Time()
+		commitTime = (*types.Timestamp)(&t)
+	}
+
+	return root, hashStr, commitTime, nil
+}
+
+// HandledFilters returns the list of filters that will be handled by the table itself
+func (dt *CommitDiffTable) HandledFilters(filters []sql.Expression) []sql.Expression {
```

Comment:
This really needs to be able to return an error, so that we can reject queries that have incompatible WHERE clauses. Might add a TODO.

Reply:
updated tests and fixed